tests/test_inference.py

```diff
@@ -1721,6 +1721,38 @@ def test_encodings_identical_results(self, genotype_encoding, num_threads):
         )
         a1.assert_data_equal(a2)
 
+    @pytest.mark.parametrize("genotype_encoding", tsinfer.GenotypeEncoding)
+    @pytest.mark.parametrize("num_threads", [0, 1, 10])
+    def test_mmap_identical_results(self, genotype_encoding, num_threads):
+        n = 27
+        m = 182
+        G, positions = get_random_data_example(n, m)
+        sample_data = tsinfer.SampleData(sequence_length=m)
+        for genotypes, position in zip(G, positions):
+            sample_data.add_site(position, genotypes)
+        sample_data.finalise()
+        # Do we get precisely the same result as the C enging with 8 bit encoding
+        # and non-mmaped storage?
+        a1 = tsinfer.generate_ancestors(sample_data)
+        with tempfile.TemporaryDirectory() as tmpdir:
+            a2 = tsinfer.generate_ancestors(
+                sample_data,
+                genotype_encoding=genotype_encoding,
+                num_threads=num_threads,
+                mmap_temp_dir=tmpdir,
+            )
+        a1.assert_data_equal(a2)
```

Comment:
Can we also test that mmap file has been removed after generating the ancestors?

Reply:
Nice, done

Reply:
Nice, done